kolibri/plugins/coach/assets/src/views/lessons/LessonSummaryPage/sidePanels/LessonResourceSelection/index.vue

```diff
@@ -0,0 +1,161 @@
+      </div>
+    </template>
+  </SidePanelModal>
+
+</template>
+
+
+<script>
+
+  import SidePanelModal from 'kolibri-common/components/SidePanelModal';
+  import { coreStrings } from 'kolibri/uiText/commonCoreStrings';
+  import { PageNames } from '../../../../../constants';
+  import SelectionIndex from './subPages/SelectionIndex';
+  import useResourceSelection from './useResourceSelection';
+  import SelectFromBookmarks from './subPages/SelectFromBookmarks.vue';
+  import SelectFromChannels from './subPages/SelectFromChannels.vue';
+  import { ResourceSelectionView } from './constants';
+
+  export default {
+    name: 'LessonResourceSelection',
+    components: {
+      SidePanelModal,
+      SelectionIndex,
+      SelectFromBookmarks,
+      SelectFromChannels,
+    },
+    setup() {
+      const { loading } = useResourceSelection();
+
+      const { saveAndFinishAction$, selectFromBookmarks$ } = coreStrings;
+
+      return {
+        loading,
+        selectFromBookmarks$,
+        saveAndFinishAction$,
+      };
+    },
+    computed: {
+      topicId() {
+        const { topicId } = this.$route.query;
+        return topicId;
+      },
+      viewId() {
+        const { viewId } = this.$route.params;
+        if (Object.values(ResourceSelectionView).includes(viewId)) {
+          return viewId;
+        }
+        return ResourceSelectionView.SELECTION_INDEX;
+      },
+      view() {
+        const componentMap = {
+          [ResourceSelectionView.SELECTION_INDEX]: {
+            title: this.$tr('manageLessonResourcesTitle'),
+            component: SelectionIndex,
+          },
+          [ResourceSelectionView.SELECT_FROM_BOOKMARKS]: {
+            title: this.selectFromBookmarks$(),
+            component: SelectFromBookmarks,
+            back: ResourceSelectionView.SELECTION_INDEX,
+          },
+          [ResourceSelectionView.SELECT_FROM_CHANNELS]: {
+            title: this.$tr('manageLessonResourcesTitle'),
+            component: SelectFromChannels,
+            back: ResourceSelectionView.SELECTION_INDEX,
+            guard: () => !!this.topicId,
+          },
+        };
+
+        return componentMap[this.viewId];
+      },
+      goBack() {
+        const { back } = this.view;
+        if (!back) {
+          return null;
+        }
+        return () => {
+          this.$router.push({ name: 'LESSON_SELECT_RESOURCES', params: { viewId: back } });
+        };
+      },
+    },
+    watch: {
+      view(newView) {
+        if (newView.guard && !newView.guard()) {
+          if (this.goBack) {
+            this.goBack();
+          } else {
+            this.$router.push({
+              name: PageNames.LESSON_SELECT_RESOURCES,
+              params: { viewId: ResourceSelectionView.SELECTION_INDEX },
+            });
+          }
+        }
+      },
+    },
+    methods: {
+      closeSidePanel() {
+        this.$router.push({
+          name: PageNames.LESSON_SUMMARY_BETTER,
+        });
+      },
+    },
```

Comment:
If the user has to explicitly save their changes then we'll need to handle the "Are you sure?" KModal to confirm losing changes.